### ISSUE TITLE
Switch to qwen2.5vl

### ISSUE THREAD

**Issue by gaoyixuan1025** [zhiyuanyou/DeQA-Score#24]
Thank you for your excellent work. We have switched to the qwen2.5vl base model for experimentation. Due to tokenizer inconsistencies, the IDs assigned to the individually segmented tokens “excellent,” “good,” “pair,” “poor,” and “bad” do not match the ‘excellent’ token within the phrase “the quality of the image is excellent.” Could this affect the soft_loss, particularly during inference when predicting logits?

**Comment by zhiyuanyou** [comment]
Yes. There are two problems. 

Problem 1: In Qwen's tokenizer, some words (like excellent and poor) can be tokenized into two tokens. 
Problem 2: In Qwen's tokenizer, "good" and " good" are two different tokens. 

You can try:

To solve Problem 1: Use new words like " great", " good", " fair", " weak", " bad". These words are one token in Qwen's tokenizer. 
To solve Problem 2: Revise the following lines (https://github.com/zhiyuanyou/DeQA-Score/blob/main/src/train/train_mem.py#L426) 
```
        for level_name in training_args.level_names:
            level_id = tokenizer(level_name)["input_ids"]
            assert len(level_id) == 2 and level_id[0] == 1
```
into the following lines. 
```
        for level_name in training_args.level_names:
            level_id = tokenizer(" " + level_name)["input_ids"]   # add " " before the word
            assert len(level_id) == 2 and level_id[0] == 1
```

Be carefully to the above two problems.

**Comment by gaoyixuan1025** [comment]
Thank you for sharing the insights about the qwen-tokenizer! We have since migrated DeQA to both Qwen2.5-7B and Qwen3-8B models for testing, and achieved promising results with improved scoring accuracy. If you're seeing suboptimal performance, it might indeed be due to potential implementation issues rather than the backbone itself.







At 2025-11-07 10:03:27, "SeanYu" ***@***.***> wrote:

vashineyu left a comment (zhiyuanyou/DeQA-Score#24)

Hi @gaoyixuan1025

I found that using capital vocab can be tokenized to one single token in qwen-tokenizer, for example: ["Excellent", "Good", "Fair", "Poor", "Bad"]

Also, how is the performance when you switch to qwen2.5-vl? Our initial try achieved around 0.91 PLCC on the KONIQ dataset, which is far away from the original mPLUG-owl2 backbone... Just wondering if the gap is sourced from the backbone or a potential implementation issue?
